### PR TITLE
New method: intersperse (121 B)

### PR DESCRIPTION
**Pull request by BabkinAleksandr**
[Ramda intersperse](http://ramdajs.com/docs/#intersperse)

`for`:
intersperse - for

`while`:
intersperse - while

`reduce` is about 100 times slower than these two

### CODE REVIEW

File: lib/intersperse/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function intersperse(separator, list) {
+  var result = []
+  for (var i = 0; i < list.length; i++) {
+    i === list.length - 1
```
Comment (Beraliv) [inline lib/intersperse/index.js]: You can iterate from 0 to n-1 and then just add last element without a condition. It must be faster not to check all the time

**Comment by Kelin2025** [comment]
Is it OK now? @Beraliv

File: lib/intersperse/index.js (hunk 1 above)
Comment (BabkinAleksandr) [reply lib/intersperse/index.js]: It have sense, but i've made tests:

whole `for`:
aleksandrbabkin_aleksandrs-macbook-pro____documents_nanoutils__zsh_

`for` with no check inside:
intersperse - for no check

Tests have been run for 100 times each with cash clearing

Comment (Kelin2025) [reply lib/intersperse/index.js]: image

**Pull request merged**